### ISSUE TITLE
automation HAT node not connecting

### ISSUE THREAD

**Issue by liltux** [shortbloke/node-red-contrib-automation-hat#12]
Hello.  I am using the automation HAT on a RPi 3 with node-red.  I recently updated my nodered install.  After the updates, Automation Hat nodes do not connect and I cannot read any inputs or send outputs.  Upgraded to Node 18 and the new Node Red 3.0.  Not sure if something is broken or I am missing something.  Running through python scripts I have no issues.

**Comment by shortbloke** [comment]
Hi, can you confirm that everything was working before your upgraded?
Assuming so, I’ll try and reproduce the issue and work out if Node or Node-Red upgrade is causing the issue.

Did you happen this try your flow after upgrading one component to help me focus on the potential problem area.

**Comment by shortbloke** [comment]
@liltux I've just setup a clean install on a Raspberry Pi 4 and tested with Node-Red v3.0.2 using node 16 and node 18.12 and everything is working.
So things to check: 
- Did you update node and node-red via the script: https://nodered.org/docs/getting-started/raspberrypi
- For example, specifying you want node v18 via the extra parameter: `bash <(curl -sL https://raw.githubusercontent.com/node-red/linux-installers/master/deb/update-nodejs-and-nodered) --node18`
- If you look at the logs from node red on startup, does it show any errors?
- If you run node-red with verbose logging: `node-red -v` do you get an output like:

```
5 Nov 15:14:56 - [info] Server now running at http://127.0.0.1:1880/
5 Nov 15:14:56 - [info] Starting flows
5 Nov 15:14:56 - [info] AutomationHAT: Initialising AutomationHATOut node
5 Nov 15:14:56 - [info] AutomationHAT: Initialising AutomationHATIn node
5 Nov 15:14:56 - [info] AutomationHAT: Initialising AutomationHATReader node
5 Nov 15:14:56 - [info] Started flows
5 Nov 15:14:56 - [info] AutomationHAT: DEBUG(HAT): Automation HAT Detected
5 Nov 15:14:56 - [info] AutomationHAT: DEBUG(HAT): handle_command: set analog threshold:0.2
5 Nov 15:14:56 - [info] AutomationHAT: DEBUG(HAT): handle_command: set adc4 threshold:undefined
5 Nov 15:14:56 - [warn] AutomationHAT: Process Exit: 1 :
5 Nov 15:14:56 - [info] AutomationHAT: Attempting Reconnect
5 Nov 15:15:01 - [info] AutomationHAT: DEBUG(HAT): Automation HAT Detected
```

**Comment by liltux** [comment]
@shortbloke, sorry for the absence and thank you for looking into this.  This is the order of actions:
1. update raspbian `sudo apt update` and `sudo apt upgrade -y`
2. update node red via the script and then again to update to --node18

I have not seen anything in the logs that stood out, that is why I came here. I will restart node-red with the -v option and see there as well, did not think to try that.  I did not start with a fresh build so I may need to do that as well.  Thanks again for you assistance!
